core/lib/dal/doc/FriProofCompressorDal.md

```diff
@@ -0,0 +1,32 @@
+# FriProofCompressorDal
+
+## Table Name
+
+proof_compression_jobs_fri
+
+
+## `status` Diagram
+
+```mermaid
+---
+title: Status Diagram
+---
+stateDiagram-v2
+[*] --> queued : insert_proof_compression_job
+queued --> in_progress : get_next_proof_compression_job
+in_progress --> successful : mark_proof_compression_job_successful
+successful --> [*] 
+in_progress --> failed : mark_proof_compression_job_failed
+failed --> [*]
+failed --> queued : requeue_stuck_jobs
+
+successful --> sent_to_server : get_least_proven_block_number_not_sent_to_server
```

Comment:
`get_least_proven_block_number_not_sent_to_server` doesn't mutate state, it should be `mark_proof_sent_to_server` instead